libs/safe/src/lib/components/widgets/map-settings/edit-layer-modal/edit-layer-modal.component.ts

```diff
@@ -239,9 +245,13 @@ export class EditLayerModalComponent
 
   /**
    * Set default layer for editor
+   *
+   * @param lastFormValue Last value triggered from the layer editor form
    */
-  private setUpLayer() {
-    if (!this.data.mapComponent) return;
+  private setUpLayer(lastFormValue?: any) {
+    if (!this.data.mapComponent) {
```

Comment:
```suggestion
    if (!this.data.mapComponent || !this.isDatasourceValid) {
```

The issue where the legend is duplicated when you remove and add the geofiled again is still present, checking if the datasource is valid before setting up the layers seems to solve it.

![Peek 2023-07-12 11-40](https://github.com/ReliefApplications/oort-frontend/assets/102038450/c87cca1d-9dfc-4f7a-9f8b-5b0cd4014e61)

Reply:
I did some more testing and that does not fix the issue :disappointed:

Reply:
Seems like adding the following line to the beginning of `onRemoveLayer` fixes this issue. However, I'm not entirely sure it won't break something else

```ts
this.visibility = false;
```

Reply:
@unai-reliefapp 
cc @matheus-relief 
I think I still got an issue when editing layers..
you can see it while running the code with distant server, on app : 69091
I created a map with a group layer
the layers still got duplicated when zooming in / out

Reply:
@AntoineRelief Fix pushed in the last commit. 
Just one little thing, on updating the layer the backed returns the previous layer properties as seen in this image.
Data sent:
![image](https://github.com/ReliefApplications/oort-frontend/assets/79921564/04c8124c-1c2d-4a85-8de1-9bbd39720fae)
Data returned:
![image](https://github.com/ReliefApplications/oort-frontend/assets/79921564/0e632aab-37c2-4f41-8e00-5a762a3cda05)

For now, on successful request Im just adding the properties sent to the backend in the map layers list item to reflect the change in the layer list, but I guess that the edit layer endpoint should return the layer with the latest properties. I can take a look at that in the backend and open the PR. In the meantime the fix added in this PR would handle any name update for the layers in the map layers list.

KR, Unai